api/controllers/v1/location/get-snapshots.js

```diff
@@ -0,0 +1,54 @@
+const ControllerService = require('../../../services/ControllerService');
+const ErrorService = require('../../../services/ErrorService');
+const RightService = require('../../../services/RightService');
+
+module.exports = async (req, res) => {
+  const CHECK_SENSITIVTY_OF_ENTRANCE_FROM_LOCATION_ID = `SELECT TE.is_sensitive
+                                                         FROM h_location HL
+                                                                JOIN t_entrance TE ON HL.id_entrance = TE.id
+                                                         WHERE HL.id = $1; `;
+  try {
+    const sensitivity = await HLocation.getDatastore().sendNativeQuery(
+      CHECK_SENSITIVTY_OF_ENTRANCE_FROM_LOCATION_ID,
+      [req.params.id]
+    );
+    const isSensitive = sensitivity.rows[0].is_sensitive;
+
+    if (isSensitive === undefined) {
+      res.serverError(
+        'A server error occured when checking the sensitivity of the entrance.'
+      );
+    }
+
+    const hasRight = isSensitive
+      ? await sails.helpers.checkRight
+          .with({
+            groups: req.token.groups,
+            rightEntity: RightService.RightEntities.ENTRANCE,
+            rightAction: RightService.RightActions.EDIT_ANY,
```

Comment:
Ce n'est pas ce right qu'il faut utiliser mais plutot cela il me semble : 
```
          rightEntity: RightService.RightEntities.ENTRANCE,
          rightAction: RightService.RightActions.VIEW_COMPLETE,
```

Reply:
Ok we will change this